File: 软件测试/第三次实验/生成二维码.java
package com.example.tests;

import java.util.regex.Pattern;
import java.util.concurrent.TimeUnit;
import org.testng.annotations.*;
import static org.testng.Assert.*;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class UntitledTestCase {
  private WebDriver driver;
  private String baseUrl;
  private boolean acceptNextAlert = true;
  private StringBuffer verificationErrors = new StringBuffer();

  @BeforeClass(alwaysRun = true)
  public void setUp() throws Exception {
    driver = new FirefoxDriver();
    baseUrl = "https://www.katalon.com/";
    driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
  }

  @Test
  public void testUntitledTestCase() throws Exception {
    driver.get("https://cli.im/text");
    driver.findElement(By.id("text-content")).click();
    driver.findElement(By.id("text-content")).clear();
    driver.findElement(By.id("text-content")).sendKeys("hello world");
    driver.findElement(By.id("click-create")).click();
    driver.findElement(By.linkText("网址")).click();
    driver.findElement(By.id("url_content")).click();
    driver.findElement(By.id("url_content")).clear();
    driver.findElement(By.id("url_content")).sendKeys("https://www.baidu.com");
    driver.findElement(By.id("click-create")).click();
    driver.findElement(By.linkText("文件")).click();
    driver.findElement(By.id("filedatacode")).click();
    driver.findElement(By.id("filedatacode")).clear();
    driver.findElement(By.id("filedatacode")).sendKeys("C:\\fakepath\\03 Wireshark_DNS_v7.0.pdf");
    driver.findElement(By.id("click-create")).click();
    driver.findElement(By.linkText("图片")).click();
    driver.findElement(By.id("filedatacode")).click();
    driver.findElement(By.id("filedatacode")).clear();
    driver.findElement(By.id("filedatacode")).sendKeys("C:\\fakepath\\v2-dcef71380baf9140dd01587f2bb396f7_r.jpg");
    driver.findElement(By.id("click-create")).click();
  }

  @AfterClass(alwaysRun = true)
  public void tearDown() throws Exception {
    driver.quit();
    String verificationErrorString = verificationErrors.toString();
    if (!"".equals(verificationErrorString)) {
      fail(verificationErrorString);
    }
  }

  private boolean isElementPresent(By by) {
    try {
      driver.findElement(by);
      return true;
    } catch (NoSuchElementException e) {
      return false;
    }
  }

  private boolean isAlertPresent() {
    try {
      driver.switchTo().alert();
      return true;
    } catch (NoAlertPresentException e) {
      return false;
    }
  }

  private String closeAlertAndGetItsText() {
    try {
      Alert alert = driver.switchTo().alert();
      String alertText = alert.getText();
      if (acceptNextAlert) {
        alert.accept();
      } else {
        alert.dismiss();
      }
      return alertText;
    } finally {
      acceptNextAlert = true;
    }
  }
}

File: 大三上/软件测试/第三次实验/生成二维码.py
# -*- coding: utf-8 -*-
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import Select
from selenium.common.exceptions import NoSuchElementException
from selenium.common.exceptions import NoAlertPresentException
import unittest, time, re

class AppDynamicsJob(unittest.TestCase):
    def setUp(self):
        # AppDynamics will automatically override this web driver
        # as documented in https://docs.appdynamics.com/display/PRO44/Write+Your+First+Script
        self.driver = webdriver.Firefox()
        self.driver.implicitly_wait(30)
        self.base_url = "https://www.katalon.com/"
        self.verificationErrors = []
        self.accept_next_alert = True
    
    def test_app_dynamics_job(self):
        driver = self.driver
        driver.get("https://cli.im/text")
        driver.find_element_by_id("text-content").click()
        driver.find_element_by_id("text-content").clear()
        driver.find_element_by_id("text-content").send_keys("hello world")
        driver.find_element_by_id("click-create").click()
        driver.find_element_by_link_text(u"网址").click()
        driver.find_element_by_id("url_content").click()
        driver.find_element_by_id("url_content").clear()
        driver.find_element_by_id("url_content").send_keys("https://www.baidu.com")
        driver.find_element_by_id("click-create").click()
        driver.find_element_by_link_text(u"文件").click()
        driver.find_element_by_id("filedatacode").click()
        driver.find_element_by_id("filedatacode").clear()
        driver.find_element_by_id("filedatacode").send_keys("C:\\fakepath\\03 Wireshark_DNS_v7.0.pdf")
        driver.find_element_by_id("click-create").click()
        driver.find_element_by_link_text(u"图片").click()
        driver.find_element_by_id("filedatacode").click()
        driver.find_element_by_id("filedatacode").clear()
        driver.find_element_by_id("filedatacode").send_keys("C:\\fakepath\\v2-dcef71380baf9140dd01587f2bb396f7_r.jpg")
        driver.find_element_by_id("click-create").click()
    
    def is_element_present(self, how, what):
        try: self.driver.find_element(by=how, value=what)
        except NoSuchElementException as e: return False
        return True
    
    def is_alert_present(self):
        try: self.driver.switch_to_alert()
        except NoAlertPresentException as e: return False
        return True
    
    def close_alert_and_get_its_text(self):
        try:
            alert = self.driver.switch_to_alert()
            alert_text = alert.text
            if self.accept_next_alert:
                alert.accept()
            else:
                alert.dismiss()
            return alert_text
        finally: self.accept_next_alert = True
    
    def tearDown(self):
        # To know more about the difference between verify and assert,
        # visit https://www.seleniumhq.org/docs/06_test_design_considerations.jsp#validating-results
        self.assertEqual([], self.verificationErrors)

if __name__ == "__main__":
    unittest.main()
